### ISSUE TITLE
NVIDIA RTX 2060 graphics card.

### ISSUE THREAD

**Issue by BlindMan64** [rescuezilla/rescuezilla#115]
HP OMEN with NVIDIA® GeForce® RTX 2060 graphics card.
Thank you for adding support for booting on EFI-only machines.
rescuezilla-1.0.6.1-64bit.iso on 2GB USB flash drive now boots to initial RescueZilla logo screen ok.
When it start to load RescueZilla GUI, screen display is lost.
Will boot safe mode to Ubuntu 20.04 LTS login prompt.

Is it possible to make GUI work with NVIDIA® GeForce® RTX 2060 graphics card?

**Comment by shasheene** [comment]
Hi BlindMan64,

Thank you for the bug report.

The NVIDIA RTX 2060 graphics card was released 1.5 years ago (January 2019). While in general very recent NVIDIA graphics cards are best supported using the closed-source `nvidia-driver` (because it often takes time for the community-supported open-source `nouveau` driver included in Rescuezilla to support for new cards), **I would still have expected Ubuntu 20.04 (released in April 2020) to provide adequate support using the open-source `nouveau` driver.** As you alluded to, the **'Safe Mode' boot menu item is intended to provide a very low resolution fallback graphical environment through the generic VESA driver** for bleeding edge hardware when drivers are not present in Rescuezilla. This `xserver-xorg-video-vesa` generic VESA driver package continues to be installed on Rescuezilla v1.0.6.1, which should be all that is required for 'Safe Mode' graphical fallback behavior to work correctly (as it does for both standard boot and 'Safe Mode' boot on my machine's relatively old NVIDIA GTX 970).

Indeed, Linux kernel 5.0 added support for the [RTX 2060 (TU106 chip)](https://www.phoronix.com/scan.php?page=news_item&px=Nouveau-RTX-2080-Ti-Linux-5.0) graphics cards way back in January 2019. Rescuezilla v1.0.6.1 64-bit provides Linux kernel 5.4 (and the installed `linux-firmware` package includes the TU106 firmware). No newer nouveau driver has yet been released:

> [linux-firmware](https://packages.ubuntu.com/focal/linux-firmware) (1.185) focal; urgency=medium
>    - nvidia: add TU102/TU104/TU106 signed firmware
>  -- Seth Forshee <seth.forshee@canonical.com>  Tue, 28 Jan 2020 15:14:26 -0600

> [xserver-xorg-video-nouveau](https://packages.ubuntu.com/focal/xserver-xorg-video-nouveau) (1:1.0.16-1) unstable; urgency=medium
>  * New upstream release.
> -- Sven Joachim <svenjoac@gmx.de>  Thu, 31 Jan 2019 21:17:23 +0100

Because of all this I would have expected an RTX 2060 to work fine with Rescuezilla v1.0.6.1 64-bit. My only suggestion would be to try switching to the proprietary driver. Ideally you would be able to do this from the command-line of the standard boot, rather than Safe Mode (which has booted with kernel options such as 'nomodeset' and 'xforcevesa' which will potentially interfere with how the proprietary NVIDIA driver is expecting to be able to configure the graphics card). I think something like the following would do it, but I haven't tested the instructions so you may hit an error (don't worry any changes only live in RAM so disappear after a reboot):

```
apt-get update
apt-get remove xserver-xorg-video-nouveau
apt-get install nvidia-driver-440-server
systemctl restart slim
```

I don't have any test environments which could debug this. **If anybody reading this has advice using Rescuezilla (or really, any Linux) with NVIDIA RTX 2060 cards, I am happy to hear it.**

(If nobody responds to this after a few days it's probably worth posting on the [Rescuezilla support forum](https://sourceforge.net/p/rescuezilla/discussion/support) as more people might have some ideas).

**Comment by Joeblacksilver** [comment]
I have the same problem. Although I have an RTX 2070.

**Comment by shasheene** [comment]
I recently found a friend with a new laptop containing an RTX 2070 GPU (and an Intel i7-10750). **We were successfully able to boot Rescuezilla v1.0.6.1 64-bit in standard mode without issues**. I tried rebooting it into Safe Mode, but it _didn't_ work for some reason (and I didn't have time to look into why at that very moment, but I'll try again soon).

@BlindMan64, @Joeblacksilver, can you provide more information about your machines? Eg, what CPU/motherboard/etc?

I hope to fix this issue for the next Rescuezilla release (scheduled for in mid-October 2020).

**Comment by BlindMan64** [comment]
Spec of my PC can be found here
https://support.hp.com/gb-en/document/c06355657

**Comment by shasheene** [comment]
@BlindMan64, @Joeblacksilver,

[Rescuezilla v2.0](https://github.com/rescuezilla/rescuezilla/releases/latest) has now been released. **Please try it and let me know how you go. If the boot still fails, try "Graphical fallback mode"**

I tried Rescuezilla on my friend's RTX 2070 laptop again. Just in case, I tried a development build that disabled the Intel onboard graphics (known as the i915 driver), and I was _still_ successful in booting, so I'm not sure exactly what is going on with your machines. Regardless, Rescuezilla v2.0 has some minor changes (such as switching from SLiM (Simple Login Manager) to LightDM (Light Display Manager) which as a more maintained project should hopefully improve reliability). Also there's a chance that this issue has been fixed in by the Ubuntu packages (but I haven't come across anything indicating this yet).

I suspect the switch from SLiM to LightDM will have improved things, but we'll see. Let me know how you guys go!

**Comment by BlindMan64** [comment]
Hi
V2, boot to splash screen ok.
Display lost when Rescuezilla starts automatically. Black screen with white line across top of screen.
Using Graphical fallback mode, boot stops at
[ok] Finished update UTMP about System Runlevel changes.
No change for 10 minutes so I pressed power button. Lots more text then displayed on screen then system powered off.

**Comment by shasheene** [comment]
Disappointing to hear there's been no improvement. I don't have any systems that reproduce the issue you've described.

One possibility would be for Rescuezilla to switch to the proprietary nvidia graphics driver instead of the open-source "nouveau" for all users, but I am very hesitant to do that: many users are open-source purists and it could cause issues for other people. Rescuezilla will also be using the new Linux kernel available through the LTS/HWE ("Hardware Enablement") project which definitely could improve things, but I believe that will be released for Ubuntu 20.04 in 6 months time, so it will be a while.

**I hopefully will be releasing Rescuezilla a standalone deb file in the not too distant future, so in the worst case you will soon be able boot eg, standard Ubuntu 20.10 as a live USB and then install Rescuezilla.**

@BlindMan64, are you running any kind of Linux operating system on the problematic machine? If so, what version?

**Comment by BlindMan64** [comment]
Only Windows10x64 installed on PC.

What should be next step in boot process after
[ok] Finished update UTMP about System Runlevel changes
?

**Comment by shasheene** [comment]
The next step in the boot process is starting the graphical session, but that step is failing. On a machine as modern as yours if the graphical session manager didn't fail the boot will be near instantaneous. The log files under /var/log/lightdm/ contain more information, but it's a bit difficult to explain how to access them.

Hopefully I'll be able to find a machine I can reproduce the issue on locally

**Comment by shasheene** [comment]
Sourceforge user [woko1754](https://sourceforge.net/p/rescuezilla/discussion/support/thread/91a2506f41/) has described a similar issue:

> I can't get the graphic running with my 'Nvidia Geforce GTX 1660 Super'-card. If I start the fallback without graphic, I can open a TTY-session. I am able to install a Nvidia-driver, but this makes no sense, without beeing able to store the changings in a persitent way.
>
> Now with my newly created account:
I can't get the graphic running with my 'Nvidia Geforce GTX 1660 Super'-card. If I start the fallback without graphic, I can open a TTY-session. I am able to install a Nvidia-driver, but this makes no sense, without beeing able to store the changings in a persitent way.
>
> Error message in Xorg.0.log is: "Screen(s) found, but none have a usable configuration"

I replied with a summary of the discussion so far, an suggested to continue the discussion here.

**Comment by woko54** [comment]
@shasheene: I tried to in install the Nvidia-driver, but I couldn't do this in the normal graphic session, because I don't get a TTY-session there. I managed to do it in a fallback-session, but then the 'systemctl restart slim' command didn't work. Something like 'no slim-device'.

**Comment by shasheene** [comment]
@woko54,

>  I managed to do it in a fallback-session, but then the 'systemctl restart slim' command didn't work. Something like 'no slim-device'.

The `sudo systemctl restart slim` was relevant for Rescuezilla v1.0.6.1 which used SLiM Simple Login Manager, but since Rescuezilla v2.0 the command to restart the display manager is `sudo systemctl restart lightdm`.

> I had the same issue with a Clonezilla-stick some weeks ago.

This indicates the issue is definitely broader than Rescuezilla, and effects Ubuntu (which Rescuezilla and some Clonezilla releases are based on). There may be other Ubuntu users who have experienced the same issue so may have some ideas, but some of the advice may not be directly applicable because both Rescuezilla and Clonezilla are pretty minimal distributions. Debugging hardware issues and drivers around Linux has always been a bit of a pain. :/

**Comment by woko54** [comment]
I installed the Nvidia-driver but unfortunately 'sudo systemctl restart lightdm' didn't change anything. I wish I 
could make the changes persistent.

[dpkg.txt](https://github.com/rescuezilla/rescuezilla/files/5403376/dpkg.txt)
[Xorg.0.log](https://github.com/rescuezilla/rescuezilla/files/5403403/Xorg.0.log)

**Comment by shasheene** [comment]
Thanks for the log files -- I wasn't expecting `sudo systemctl restart lightdm` to change the outcome (whether positive or negative), I was just pointing out that that is the new command to restart the display manager.

**Comment by shasheene** [comment]
I should also add, the Graphical Fallback Mode is running with kernel boot options like "nomodeset" which might effect nvidia-driver (I'm not sure).

Sorry for the lack of solution so far, I'll spend some more time looking into this more detail soon.

**Comment by shasheene** [comment]
@woko54 (and any others),

**I have created a test build of Rescuezilla v2.0 on Ubuntu 20.10 (Groovy), rather than Ubuntu 20.04 (Focal). Please download it and check if your problematic machines boot: [rescuezilla.amd64.groovy-test-1.2020-10-29.iso](https://drive.google.com/file/d/1FX-nsMtSY_8h9l6-Ski8AamYdIoceQRH/view?usp=sharing)**

The only relevant change is using a newer Linux kernel version 5.8.0-26, rather than Linux kernel version 5.4.0-51 that was used in Rescuezilla v2.0. I was looking through the Linux kernel release notes, and Linux kernel [v5.6](https://kernelnewbies.org/Linux_5.6) **has an item for the nouveau firmware indicating it has added "TU10x graphics engine support (TU11x pending)"**, with modifications that seem to indicate the problem you all have described has been resolved.

If this build works for you, from Rescuezilla v2.1 they'll be a build based on Ubuntu 20.10 (Groovy) using the nouveau open-source firmware. Otherwise the new Ubuntu 20.10 (Groovy) build will be using the proprietary nvidia-driver (which will definitely work, but many users will prefer to avoid using proprietary software).

**Comment by woko54** [comment]
This worked for me out of the box !

Thank you so much

**Comment by shasheene** [comment]
Woohoo! I'm happy to hear it!

@woko54, please **don't** use that version for important data: upgrading the operating system often makes many internal changes, and **I have not yet done thorough testing**, I just made that version to confirm whether Ubuntu 20.10 (Groovy) would solve this NVidia problem.

I will be releasing Rescuezilla v2.1 in 1 or 2 weeks and I will have a version based on Ubuntu 20.10 "Groovy" that's ready to be widely used.

**Comment by BlindMan64** [comment]
rescuezilla.amd64.groovy-test-1.2020-10-29.iso booted to desktop offering Backup and Restore options. Look forward to trying Rescuezilla v2.1.

Thank You.

**Comment by shasheene** [comment]
Fixed in [Rescuezilla v2.1](https://github.com/rescuezilla/rescuezilla/releases/latest).

I am removing the link to the groovy test image because Rescuezilla v2.1 supercedes it.
